### PR TITLE
Feat: Route and Resource Loading Add --path to beacon.install generator

### PR DESCRIPTION
**Pull request by edborsa**
https://github.com/BeaconCMS/beacon/issues/306

### CODE REVIEW

File: lib/beacon/types/site.ex [hunk 1]
```diff
@@ -27,6 +27,8 @@ defmodule Beacon.Types.Site do
   def valid?(site) when site in ["", nil, true, false], do: false
 
   def valid?(site) when is_binary(site) do
+    # TODO: Should we add the Regex.match? from the line bellow?
```
Comment (leandrocp) [inline lib/beacon/types/site.ex:30]: > Should we add the Regex.match? from the line bellow?

Yes.

File: lib/beacon/types/site.ex [hunk 2]
```diff
@@ -44,6 +46,14 @@ defmodule Beacon.Types.Site do
     not String.starts_with?(site, "beacon_")
   end
 
+  def valid_path?(path) when is_atom(path) do
+    valid_name?(Atom.to_string(path))
```
Comment (leandrocp) [inline lib/beacon/types/site.ex:50]:
```suggestion
    path |> Atom.to_string() |> valid_name?()
```

File: lib/beacon/types/site.ex [hunk 3]
```diff
@@ -44,6 +46,14 @@ defmodule Beacon.Types.Site do
...
+  end
+
+  def valid_path?(path) when is_binary(path) do
+    String.starts_with?(path, "/")
```
Comment (leandrocp) [inline lib/beacon/types/site.ex:54]: 👍🏻

File: lib/mix/tasks/install.ex [hunk 4]
```diff
@@ -334,12 +342,30 @@ defmodule Mix.Tasks.Beacon.Install do
   end
 
   defp validate_options!(options) do
-    site = String.to_atom(options[:site])
+    site = options |> Keyword.get(:site, "false") |> String.to_atom()
+    path = options |> Keyword.get(:path, "/#{site}")
```
Comment (leandrocp) [inline lib/mix/tasks/install.ex:346]: Actually you can just use the value from `options` directly and let `Beacon.Types.Site` do the actual validation.

```suggestion
```

And then:

```elixir
!Beacon.Types.Site.valid?(options[:site]) ...
```

**Review by leandrocp — CHANGES_REQUESTED** [review]
Just a few comments. Thanks

File: lib/beacon/types/site.ex (hunk 1 above)
Comment (edborsa) [reply lib/beacon/types/site.ex]: https://github.com/BeaconCMS/beacon/pull/387/commits/32737129a17f34cce212c5af479936281859c348

File: lib/beacon/types/site.ex (hunk 2 above)
Comment (edborsa) [reply lib/beacon/types/site.ex]: https://github.com/BeaconCMS/beacon/pull/387/commits/d7eeb81b6d9efae364809cde9908932dd530f867

File: lib/mix/tasks/install.ex (hunk 4 above)
Comment (edborsa) [reply lib/mix/tasks/install.ex]: https://github.com/BeaconCMS/beacon/pull/387/commits/162a51ad319d3ed83e7741be375e188e849738ed

**Review by leandrocp — APPROVED** [review]

**Comment by leandrocp** [comment]
❤️

**Pull request merged**